adrs/0153-checkout-v2.md

```diff
@@ -179,6 +183,33 @@ A better solution is:
 
 Given a source file path, walk up the directories until the first `.git/config` is found. Check if it matches the self repo (`url = https://github.com/OWNER/REPO`). If not, drop the source file path.
 
+
+### User name and email
+
+The `user-name` and `user-email` can be used to set the user name and email in the local git config.
+
+This is useful for the command `git commit`, which requires a user name and email to be configured.
+
+The example uses the GITHUB_TOKEN to push a commit to the server:
+
+```yaml
+jobs:
+  my-job:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          user-name: github-actions
+          user-email: github-actions-bot@users.noreply.github.com
```

Comment:
I am unsure off the top of my head, but is there a way in your actions file to reference the user and email on the commit being built from or the action triggering user?  As I mentioned in #13 I would like to be able to amend the commit or make a new one without having different commit ownership.

If not it would be awesome if these were just defaulted to that with these option here to override. Or is there a reason that is not a good idea?

Reply:
On push it looks like `${{ github.event.head_commit.author.name }}` and `${{ github.event.head_commit.author.email }}` can be used.

Haven't looked at values on pull_request. I dumped the context using this:

```yaml
on: push
jobs:
  build:
    runs-on: [ubuntu-latest]
    steps:
      - run: |
          echo "$GITHUB_CONTEXT"
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
```

I'm hesitant to add any default since it would regress folks who are setting and relying on global config user name. If we make a new major version (v3), then would be an opportunity to break compat.

Reply:
Makes sense, as long as there is an easy way to achieve this!

Reply:
@wesleytodd I believe you can read the context/json and set outputs in the step before using checkout. However, this seems unnecessarily complex. @ericsciple, what do you think about supporting a single `user` field that expects either `username <useremail>` or `auto`? Naturally, `auto` would imply the behaviour expected by @wedleytodd (and me).

Reply:
Does GitHub associate `github-actions-bot@users.noreply.github.com` with *github-actions[bot]*? Otherwise, `github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>` might be a better default as I [mentioned in #13](https://github.com/actions/checkout/issues/13#issuecomment-724415212).

@wesleytodd Amending a commit does not change the author unless `--reset-author` or `--author=<author>` is set. Only the commiter is changed but I think that is fine in most cases since it reflects what is actually happening.